src/main/java/org/springframework/data/elasticsearch/core/mapping/IndexInformation.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.elasticsearch.core.mapping;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.common.settings.Settings;
+
+/**
+ * Immutable object that holds information(name, settings, mappings, aliases) about an Index
+ *
+ * @author George Popides
+ * @since 4.2
+ */
+public class IndexInformation {
+	private final String name;
+	private final Settings settings;
+	private final MappingMetadata mappings;
+	private final List<AliasMetadata> aliases;
+
```

Comment:
settings, mapping and alias information should probably be annotated as `@Nullable` as they need not be set ion an index.

In Spring Data Elasticsearch every parameter, property or return type that is not annotated with `org.springframework.lang.Nullable` is regarded as being non-null.